File: app/src/main/java/co/fddittmar/planbuilder/presenter/MainPresenter.java
package co.fddittmar.planbuilder.presenter;

import android.util.Log;

import java.util.List;

import co.fddittmar.planbuilder.contracts.MainContract;
import co.fddittmar.planbuilder.data.ProgramRepository;
import co.fddittmar.planbuilder.data.model.Program;

import static android.app.Activity.RESULT_OK;

/**
 * Presenter for the MainActivity. It takes a reference to a View and a Repository.
 */

public class MainPresenter implements MainContract.Actions {
    MainContract.View view;
    ProgramRepository repository;

    public MainPresenter(MainContract.View view, ProgramRepository repository) {
        this.view = view;
        this.repository = repository;
    }

    @Override
    public List<Program> getAllPrograms() {
        if(repository == null)
            Log.d("DEBUG", "Null repository");
        return repository.fetchAllPrograms();
    }

    @Override
    public void checkNewProgramText(List<Program> programList) {
        if(programList.size() > 0){
            view.hideNewProgramText();
        }else {
            view.showNewProgramText();
        }
    }

    @Override
    public void onAddBtnClicked() {
        view.openNewProgramActivity();
    }

    @Override
    public void onItemProgramClicked(Program program) {
        view.openProgramDetailActivity(program);
    }

    @Override
    public void checkActivityResult(int resultCode) {
        if (resultCode == RESULT_OK) {
            view.newProgramAdded();
        }
    }

    @Override
    public void deleteProgram(long program_id) {
        repository.deleteProgram(program_id);
    }

}

File: app/src/main/java/co/fddittmar/planbuilder/presenter/MainPresenter.kt
package co.fddittmar.planbuilder.presenter

import android.app.Activity.RESULT_OK
import co.fddittmar.planbuilder.contracts.MainContract
import co.fddittmar.planbuilder.data.ProgramRepository
import co.fddittmar.planbuilder.data.model.Program

/**
 * Presenter for the MainActivity. It takes a reference to a View and a Repository.
 */

class MainPresenter(internal var view: MainContract.View, internal var repository: ProgramRepository?) : MainContract.Actions {

    override val allPrograms: ArrayList<Program>
        get() {
            return repository!!.fetchAllPrograms()
        }

    override fun checkNewProgramText(programList: List<Program>) {
        if (programList.isNotEmpty()) {
            view.hideNewProgramText()
        } else {
            view.showNewProgramText()
        }
    }

    override fun onAddBtnClicked() {
        view.openNewProgramActivity()
    }

    override fun onItemProgramClicked(program: Program) {
        view.openProgramDetailActivity(program)
    }

    override fun checkActivityResult(resultCode: Int) {
        if (resultCode == RESULT_OK) {
            view.newProgramAdded()
        }
    }

    override fun deleteProgram(program_id: Long) {
        repository!!.deleteProgram(program_id)
    }

}
